### PR TITLE
terraform support for inventory source

### PR DESCRIPTION
**Pull request by apoorv-garg**

### CODE REVIEW

**Comment by apoorv-garg** [comment]
Please feel free to review, while I add acceptance tests

**Comment by apoorv-garg** [comment]
@seansain to write acceptance tests, we would need to hardcode an AWS role ARN, or create one dynamically using cloudformation, in which case we would need to hardcode some aws credentials. Is there a prior art for this ? I can't find any existing tests for the polling source.

Also please let me know if acceptance tests are an absolute must here. We are running on a very tight GA timeline, so another option would be to add tests post that (1st week of september), and work with QE to release these without acceptance tests meanwhile.

**Comment by seansain** [comment]
Ohh, that's right. I forgot polling source doesn't have acceptance tests yet. We can hard code a role ARN if we use it as environment variable. That would be ideal I think.

However, QE verification for the tight GA timeline should suffice for now.

**Comment by seansain** [comment]
LGTM, but I think @maimaisie or @vsinghal13 could provide a better review.

**Comment by seansain** [comment]
just merged #70 that should fix the lint error

File: website/docs/r/aws_inventory_source.html.markdown [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+## Argument reference
+
+In addition to the common properties, the following arguments are supported:
+
+ - `content_type` - (Required) The content-type of the collected data. This has to be `AwsInventoryPath` for AWS Inventory source.
+ - `scan_interval` - (Required) Time interval in milliseconds of scans for new data. The minimum value is 1000 milliseconds. Currently this value is not respected.
+ - `paused` - (Required) When set to true, the scanner is paused. To disable, set to false.
+ - `authentication` - (Required) Authentication details to access AWS `Describe*` APIs.
+     + `type` - (Required) Must be either `S3BucketAuthentication` or `AWSRoleBasedAuthentication`
```
Comment (vsinghal13) [inline website/docs/r/aws_inventory_source.html.markdown:50]: Maybe we should modify this to have only `AWSRoleBasedAuthentication` as `S3BucketAuthentication` is not supported for AWS Inventory source?

File: sumologic/resource_sumologic_generic_polling_source.go [hunk 2]
```diff
@@ -278,29 +293,35 @@ func getPollingTagFilters(d *schema.ResourceData) []TagFilter {
...
 	auths := d.Get("authentication").([]interface{})
 	authSettings := PollingAuthentication{}
 
 	if len(auths) > 0 {
 		auth := auths[0].(map[string]interface{})
 		switch authType := auth["type"].(string); authType {
 		case "S3BucketAuthentication":
+			if d.Get("content_type").(string) == "AwsInventory" {
+				return authSettings, errors.New(
+					fmt.Sprintf("[ERROR] Unsupported authType: %v for AwsInventory source", authType))
```
Comment (maimaisie) [inline sumologic/resource_sumologic_generic_polling_source.go:306]: do we block source creation on the backend when the auth type is bucket auth for inventory sources? if so, I'm wondering if we need to repeat the validation in the tf layer

File: website/docs/r/aws_inventory_source.html.markdown (hunk 1 above)
Comment (apoorv-garg) [reply website/docs/r/aws_inventory_source.html.markdown]: Good catch, fixed.

File: sumologic/resource_sumologic_generic_polling_source.go (hunk 2 above)
Comment (apoorv-garg) [reply sumologic/resource_sumologic_generic_polling_source.go:306]: We don't actually. We are, however, considering adding support in the backend, so if and when that is done, I will remove this check from here. OTOH, if we end up deciding not to support bucket auth in backend, we will make that check in the backend in addition to this one.

**Comment by seansain** [comment]
@apoorv-garg can you update the Changelog with info on the two new PRs before merging?

**Comment by apoorv-garg** [comment]
> @apoorv-garg can you update the Changelog with info on the two new PRs before merging?

Done

**Comment by vsinghal13** [comment]
@apoorv-garg could you please add acceptance tests for Xray and Inventory sources? Please refer to #72 for templates.

**Comment by apoorv-garg** [comment]
Sure, will try to do it by EOW

**Pull request merged**